components/casestudies/Projects.tsx

```diff
@@ -0,0 +1,48 @@
+import Image from "next/image";
+import React from "react";
+
+type Props = {
+  title: string;
+  description: string;
+  project: string;
+  bg: string;
+};
+
+const Projects = (props: Props) => {
+  const bgColor = `bg-${props.bg}`;
+
+  return (
```

Comment:
This is why your bg color isn't working. It's a fundamental misunderstanding of what tailwind is doing. 

Tailwind doesn't include each possible tailwind class or option as CSS when you write your code. Instead it scans all of your code looking for potential tailwind classes and then adds them to a list of classes to generate CSS for.

So when you write "bg-${props.bg}" it doesn't see any relevant class, and whatever you pass into that at runtime won't be checked or generated by tailwind unless you're using the JIT compiler in the page itself.

You can instead use a 'style' property to overwrite the background if you insist on doing it this way. Or you can write the potential background colors in tailwind ahead of time.

with style:
```js
<div style={{backgroundColor:props.bg}}>
```

with tailwind:
```js
const bgBlue = "bg-blue-500";
const bgRed = "bg-red-500";
const currentColor = "";
if(props.bg==="red") currentColor = bgRed;
if(props.bg==="blue") currentColor = bgBlue;
```
This way tailwind will see the classes ahead of time and generate the CSS needed for those to work. If you want it dynamic then you must use style- if you have select number of options you can use them ahead of time like that and tailwind will make them for you.

Reply:
@brandonetter 
I was using colors from my tailwind config, shouldn't those colors have worked? Some of the colors were working well others were not. I tried each method that you outlined and was not able to get the tailwind one to work so I went with style.